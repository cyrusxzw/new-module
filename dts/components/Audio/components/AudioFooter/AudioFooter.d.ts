import React from 'react';
import type { AudioFooterProps } from './AudioFooter.types';
declare const AudioFooter: React.NamedExoticComponent<AudioFooterProps>;
export { AudioFooter };
