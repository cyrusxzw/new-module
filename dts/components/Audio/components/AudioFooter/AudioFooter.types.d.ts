import type { AudioCopy, ProgressColor } from '../../Audio.types';
declare type OnPlayPauseButtonClick = () => void;
declare type OnSeekBackwardButtonClick = () => void;
declare type OnSeekForwardButtonClick = () => void;
declare type AudioFooterProps = {
  audioUrl?: string;
  copy?: AudioCopy;
  duration?: number;
  isLoading?: boolean;
  isPlaying?: boolean;
  onSeekBackwardButtonClick?: OnSeekBackwardButtonClick;
  onSeekForwardButtonClick?: OnSeekForwardButtonClick;
  onPlayPauseButtonClick?: OnPlayPauseButtonClick;
  progress?: number;
  progressColor?: ProgressColor;
};
export type {
  AudioFooterProps,
  OnPlayPauseButtonClick,
  OnSeekBackwardButtonClick,
  OnSeekForwardButtonClick,
};
