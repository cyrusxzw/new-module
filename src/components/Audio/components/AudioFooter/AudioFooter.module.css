@import 'styles/index';

.base {
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  padding-bottom: 35px;
  margin-top: 20px;

  @media (--viewport-md) {
    flex-direction: row;
    padding-bottom: 0;
  }
}

.time {
  margin: auto 0;
  font-size: rem(15px);
  transition: color 400ms var(--easing-ease-out-cubic);

  &.disabled {
    color: var(--color-dark-disabled);
  }

  @media (--viewport-xs-sm-only) {
    position: absolute;
    bottom: 0;
    left: 0;
    font-size: rem(14px);
  }

  @media (--viewport-md) {
    width: calc(100% * 1 / 3);
  }
}

.controls {
  display: flex;
  flex-direction: row-reverse;
  align-items: center;
  justify-content: center;

  @media (--viewport-md) {
    width: calc(100% * 1 / 3);
  }
}

.download {
  margin: auto 0;
  text-align: right;

  @media (--viewport-md) {
    width: calc(100% * 1 / 3);
  }

  @media (--viewport-xs-sm-only) {
    position: absolute;
    right: 0;
    bottom: 0;
    font-size: rem(14px);
  }
}

.downloadButton {
  font-size: rem(15px);
  transition: color 200ms var(--easing-ease-out-cubic);

  @media (--viewport-xs-sm-only) {
    font-size: rem(14px);
  }

  .downloadButtonIcon {
    transition: fill 200ms var(--easing-ease-out-cubic);
  }

  &:hover {
    .orange & {
      color: var(--color-highlight-orange);
    }

    .green & {
      color: var(--color-highlight-green);
    }

    .blue & {
      color: var(--color-highlight-blue);
    }
  }

  &:hover .downloadButtonIcon {
    .orange & {
      fill: var(--color-highlight-orange);
    }

    .green & {
      fill: var(--color-highlight-green);
    }

    .blue & {
      fill: var(--color-highlight-blue);
    }
  }
}
