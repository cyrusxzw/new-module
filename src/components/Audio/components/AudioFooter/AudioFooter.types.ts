import type { AudioCopy, ProgressColor } from '../../Audio.types';

type OnPlayPauseButtonClick = () => void;
type OnSeekBackwardButtonClick = () => void;
type OnSeekForwardButtonClick = () => void;

type AudioFooterProps = {
  audioUrl?: string;
  copy?: AudioCopy;
  duration?: number;
  isLoading?: boolean;
  isPlaying?: boolean;
  onSeekBackwardButtonClick?: OnSeekBackwardButtonClick;
  onSeekForwardButtonClick?: OnSeekForwardButtonClick;
  onPlayPauseButtonClick?: OnPlayPauseButtonClick;
  progress?: number;
  progressColor?: ProgressColor;
};

export type {
  AudioFooterProps,
  OnPlayPauseButtonClick,
  OnSeekBackwardButtonClick,
  OnSeekForwardButtonClick,
};
