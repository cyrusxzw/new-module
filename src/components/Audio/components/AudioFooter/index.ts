import { AudioFooter } from './AudioFooter';

export { AudioFooter };
