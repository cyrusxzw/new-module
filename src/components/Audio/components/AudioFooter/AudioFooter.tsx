import React from 'react';
import cx from 'classnames';
import moment from 'moment';
import { HYPERLINK_STYLE_TYPES } from '~/constants';
import { Hyperlink } from '~/components/Hyperlink';
import { Icon } from '~/components/Icon';
import { PausePlayButton } from '~/components/Audio/components/PausePlayButton';
import { SeekButton } from '~/components/Audio/components/SeekButton';
import type { AudioFooterProps } from './AudioFooter.types';
import styles from './AudioFooter.module.css';

const AudioFooter = React.memo<AudioFooterProps>(
  ({
    audioUrl,
    copy = {},
    duration,
    isLoading,
    isPlaying,
    onSeekBackwardButtonClick,
    onSeekForwardButtonClick,
    onPlayPauseButtonClick,
    progress,
    progressColor = 'orange',
  }) => {
    return (
      <footer className={cx(styles.base, styles[progressColor])}>
        <time className={cx(styles.time, { [styles.disabled]: isLoading })}>
          {moment.utc(progress * 1000).format('mm:ss')} /{' '}
          {moment.utc(duration * 1000).format('mm:ss')}
        </time>
        <div className={styles.controls}>
          <SeekButton
            copy={{
              title: copy.seekForward,
            }}
            direction="forward"
            isLoading={isLoading}
            onClick={onSeekForwardButtonClick}
            progressColor={progressColor}
          />
          <PausePlayButton
            copy={{
              loading: copy.loading,
              pause: copy.pause,
              play: copy.play,
            }}
            isLoading={isLoading}
            isPlaying={isPlaying}
            onClick={onPlayPauseButtonClick}
            progressColor={progressColor}
          />
          <SeekButton
            copy={{
              title: copy.seekBackward,
            }}
            direction="backward"
            isLoading={isLoading}
            onClick={onSeekBackwardButtonClick}
            progressColor={progressColor}
          />
        </div>
        <div className={styles.download}>
          <Hyperlink
            className={styles.downloadButton}
            hasTargetInNewWindow={true}
            isDownload={true}
            style={HYPERLINK_STYLE_TYPES.EXTERNAL_NO_ICON_TEXT_LINK}
            title={copy.downloadTitle}
            url={audioUrl}
          >
            {`${copy.downloadLabel} `}
            <Icon
              className={styles.downloadButtonIcon}
              height={13}
              name="download"
              width={13}
            />
          </Hyperlink>
        </div>
      </footer>
    );
  },
);

export { AudioFooter };
