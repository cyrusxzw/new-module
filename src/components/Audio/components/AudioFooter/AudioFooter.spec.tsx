import React from 'react';
import { render } from '@testing-library/react';
import { AudioFooter } from './AudioFooter';

describe('<Audio.AudioFooter />', () => {
  it('should be defined', () => {
    expect(AudioFooter).toBeDefined();
  });

  it('should render base component correctly', () => {
    const { container } = render(<AudioFooter audioUrl="/" />);

    expect(container).toMatchSnapshot();
  });
});
