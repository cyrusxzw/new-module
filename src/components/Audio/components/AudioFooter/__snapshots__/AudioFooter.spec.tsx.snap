// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Audio.AudioFooter /> should render base component correctly 1`] = `
<div>
  <footer
    class="base orange"
  >
    <time
      class="time"
    >
      Invalid date
       /
       
      Invalid date
    </time>
    <div
      class="controls"
    >
      <button
        class="base dark base orange forward"
        type="button"
      >
        <svg
          aria-labelledby="seek-1"
          class="base seek dark icon"
          data-testid="data-testid-Icon"
          focusable="false"
          height="20"
          role="img"
          style="height: 20px; width: 20px;"
          tabindex="-1"
          viewBox="0 0 20 12"
          width="20"
        >
          <title
            id="seek-1"
          >
            seek
          </title>
          <g>
            <path
              d="M19.996,11.991 L9.999,5.990 L19.996,-0.012"
            />
            <path
              d="M9.996,11.991 L-0.001,5.990 L9.996,-0.012"
            />
          </g>
        </svg>
      </button>
      <button
        class="base dark base orange"
        type="button"
      >
        <svg
          aria-labelledby="play-2"
          class="base play dark iconPlay"
          data-testid="data-testid-Icon"
          focusable="false"
          height="15"
          role="img"
          style="height: 15px; width: 15px;"
          tabindex="-1"
          viewBox="0 0 8 11"
          width="15"
        >
          <title
            id="play-2"
          >
            play
          </title>
          <g>
            <polygon
              class="play"
              points="0,0 8,5.5 0,11"
            />
          </g>
        </svg>
        <svg
          aria-labelledby="pause-3"
          class="base pause dark iconPause hidden"
          data-testid="data-testid-Icon"
          focusable="false"
          height="15"
          role="img"
          style="height: 15px; width: 15px;"
          tabindex="-1"
          viewBox="0 0 9 10"
          width="15"
        >
          <title
            id="pause-3"
          >
            pause
          </title>
          <g>
            <polygon
              class="play"
              points="0,0 3,0 3,10 0,10 0,0"
            />
            <polygon
              class="play"
              points="6,0 9,0 9,10 6,10 6,0"
            />
          </g>
        </svg>
      </button>
      <button
        class="base dark base orange"
        type="button"
      >
        <svg
          aria-labelledby="seek-4"
          class="base seek dark icon"
          data-testid="data-testid-Icon"
          focusable="false"
          height="20"
          role="img"
          style="height: 20px; width: 20px;"
          tabindex="-1"
          viewBox="0 0 20 12"
          width="20"
        >
          <title
            id="seek-4"
          >
            seek
          </title>
          <g>
            <path
              d="M19.996,11.991 L9.999,5.990 L19.996,-0.012"
            />
            <path
              d="M9.996,11.991 L-0.001,5.990 L9.996,-0.012"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="download"
    >
      <a
        class="base left dark downloadButton"
        download=""
        href="/"
        target="_blank"
      >
        undefined 
        <svg
          aria-labelledby="download-5"
          class="base download dark downloadButtonIcon"
          data-testid="data-testid-Icon"
          focusable="false"
          height="13"
          role="img"
          style="height: 13px; width: 13px;"
          tabindex="-1"
          viewBox="0 0 11 14"
          width="13"
        >
          <title
            id="download-5"
          >
            download
          </title>
          <g>
            <path
              d="M10.992,6.125 L6.125,10.782 L6.125,0.001 L4.848,0.001 L4.848,10.768 L-0.004,6.125 L-0.004,7.819 L5.495,13.011 L10.992,7.819 L10.992,6.125 Z"
            />
            <path
              d="M-0.000,12.999 L11.000,12.999 L11.000,14.000 L-0.000,14.000 L-0.000,12.999 Z"
            />
          </g>
        </svg>
      </a>
    </div>
  </footer>
</div>
`;
